contracts/staking/ServiceStakingToken.sol

```diff
@@ -0,0 +1,116 @@
+
+import {ServiceStakingBase} from "./ServiceStakingBase.sol";
+import {SafeTransferLib} from "../utils/SafeTransferLib.sol";
+import "../interfaces/IToken.sol";
+
+// Service Registry Token Utility interface
+interface IServiceTokenUtility {
+    /// @dev Gets the service security token info.
+    /// @param serviceId Service Id.
+    /// @return Token address.
+    /// @return Token security deposit.
+    function mapServiceIdTokenDeposit(uint256 serviceId) external view returns (address, uint96);
+}
+
+/// @dev The token does not have enough decimals.
+/// @param token Token address.
+/// @param decimals Number of decimals.
+error NotEnoughTokenDecimals(address token, uint8 decimals);
+
+/// @dev The staking token is wrong.
+/// @param expected Expected staking token.
+/// @param provided Provided staking token.
+error WrongStakingToken(address expected, address provided);
+
+/// @dev Received lower value than the expected one.
+/// @param provided Provided value is lower.
+/// @param expected Expected value.
+error LowerThan(uint256 provided, uint256 expected);
+
+/// @title ServiceStakingToken - Smart contract for staking a service by its owner when the service has an ERC20 token as the deposit
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author Andrey Lebedev - <andrey.lebedev@valory.xyz>
+/// @author Mariapia Moscatiello - <mariapia.moscatiello@valory.xyz>
+contract ServiceStakingToken is ServiceStakingBase {
+    // ServiceRegistryTokenUtility address
+    address public immutable serviceRegistryTokenUtility;
+    // Security token address for staking corresponding to the service deposit token
+    address public immutable stakingToken;
+
+    /// @dev ServiceStakingToken constructor.
+    /// @param _maxNumServices Maximum number of staking services.
+    /// @param _rewardsPerSecond Staking rewards per second (in single digits).
+    /// @param _minStakingDeposit Minimum staking deposit for a service to be eligible to stake.
+    /// @param _livenessRatio Liveness ratio: number of nonces per second (in 18 digits).
+    /// @param _serviceRegistry ServiceRegistry contract address.
+    /// @param _serviceRegistryTokenUtility ServiceRegistryTokenUtility contract address.
+    /// @param _stakingToken Address of a service staking token.
+    constructor(
+        uint256 _maxNumServices,
+        uint256 _rewardsPerSecond,
+        uint256 _minStakingDeposit,
+        uint256 _livenessRatio,
+        address _serviceRegistry,
+        address _serviceRegistryTokenUtility,
+        address _stakingToken
+    )
+        ServiceStakingBase(_maxNumServices, _rewardsPerSecond, _minStakingDeposit, _livenessRatio, _serviceRegistry)
+    {
+        // Initial checks
+        if (_stakingToken == address(0) || _serviceRegistryTokenUtility == address(0)) {
+            revert ZeroAddress();
+        }
+
+        stakingToken = _stakingToken;
+        serviceRegistryTokenUtility = _serviceRegistryTokenUtility;
```

Comment:
Should also check that the staking token matches the one used for security deposit? I mean in principle one could break this symmetry. But I think we should then have a different contract. Symmetry is more "expected"

Reply:
We check that for each service itself. ServiceRegistryTokenUtility is not bound to a specific custom token.